### DISCUSSION TITLE
Why do the exportDOM and importDOM only export\import partial style attribute, not the full style?

### DISCUSSION THREAD

**Discussion by SD-Gaming** [facebook/lexical#6561] [Q&A] [answered]
In the LexicalParagraphNode class implementation's importDOM method, only the HTMLElement 's style properties, `textAlign` and `textIndent`, are received; all other properties are ignored.

The implementation of the exportDOM method only exports the `textAlign` `textIndent` style property and `dir`. all other properties are ignored, too.

and the LexicalTextNode also follows the above rule.

The result is that when I export an HTML string with $generateHtmlFromNodes and process Lexical -> HTML, many styles are lost.
also when I convert HTML -> Lexical.

My problem is, what's the reason did the community design this?

**Comment by ivailop7** [comment, marked as answer]
Explanation is in the docs:

https://lexical.dev/docs/concepts/serialization#handling-extended-html-styling
